examples/md-flexible/tests/domainDecomposition/MixedBoundaryConditionTest.cpp

```diff
@@ -41,43 +41,53 @@ void initializeAutoPasContainer(RegularGridDecomposition::SharedAutoPasContainer
+ * - cases where the particle is in a corner, including
+ *   + reflective/reflective
+ *   + reflective/periodic
+ *   + reflective/reflective/periodic
+ * - cases for both upper and lower boundaries
  */
 TEST_F(MixedBoundaryConditionTest, testSimpleReflection) {
-  std::vector<std::string> arguments = {"md-flexible", "--yaml-filename",
-                                        std::string(YAMLDIRECTORY) + "reflectionTest.yaml"};
-
-  char *argv[3] = {arguments[0].data(), arguments[1].data(), arguments[2].data()};
-
-  MDFlexConfig configuration(3, argv);
-
-  std::array<double, 3> localBoxMin = configuration.boxMin.value;
-  std::array<double, 3> localBoxMax = configuration.boxMax.value;
-
-  RegularGridDecomposition domainDecomposition(configuration.boxMin.value, configuration.boxMax.value,
-                                               configuration.subdivideDimension.value, configuration.cutoff.value,
-                                               configuration.verletSkinRadius.value, configuration.reflWidth.value,
-                                               configuration.boundaryOption.value);
+  // initialise AutoPas container & domainDecomposition
+  const std::array<double,3> boxMin = {0.,0.,0.};
+  const std::array<double,3> boxMax = {5.,5.,5.};
+  const std::array<double,3> boxLength = {boxMax[0] - boxMin[0],boxMax[1] - boxMin[1],boxMax[2] - boxMin[2]};
+  const std::array<bool,3> subdivideDimension = {true,true,true};
+  const double cutoffWidth = 2.;
+  const double skinWidth = 0.2;
+  const std::array<options::BoundaryTypeOption,3> boundaryConditions =
+      {options::BoundaryTypeOption::periodic,options::BoundaryTypeOption::reflective,options::BoundaryTypeOption::reflective};
+
+  RegularGridDecomposition domainDecomposition(boxMin, boxMax, subdivideDimension,
+                                               cutoffWidth, skinWidth,boundaryConditions);
 
   auto autoPasContainer = std::make_shared<autopas::AutoPas<ParticleType>>(std::cout);
 
-  initializeAutoPasContainer(autoPasContainer, configuration);
+  //initializeAutoPasContainer(autoPasContainer, configuration);
+  autoPasContainer->setBoxMin(boxMin);
+  autoPasContainer->setBoxMax(boxMax);
+  autoPasContainer->setCutoff(cutoffWidth);
+  autoPasContainer->setVerletSkin(skinWidth);
+  autoPasContainer->init();
 
-  std::vector<std::vector<double>> particlePositions = {
+  const std::vector<std::array<double, 3>> particlePositions = {
       {1.0,0.05,1.0}, {2.0,0.05,1.0}, {1.0,0.05,0.05}, {2.0,0.05,0.05}, {4.95,0.05,0.05},
       {4.0,4.95,4.0}, {3.0,4.95,4.0}, {4.0,4.95,4.95}, {3.0,4.95,4.95}, {0.05,4.95,4.95}
   };
-  std::vector<std::vector<double>> particleVelocities = {
+  const std::vector<std::array<double, 3>> particleVelocities = {
       {0.0,-1.0,1.0}, {0.0,1.0,1.0}, {0.0,-1.0,-0.5}, {0.0,-1.0,0.5}, {1.0,-0.5,-0.2},
       {0.0,1.0,-1.0}, {0.0,-1.0,-1.0}, {0.0,1.0,0.5}, {0.0,1.0,-0.5}, {-1.0,0.5,0.2}
   };
```

Comment:
This one is purely optional and up to your liking:
Here you have one test, testing 10 cases. That is a nice example where you could use value parametrized tests to generate a test for each of the cases! The advantage would be that if something fails you could see directly which cases fail. Also, the tests can be executed concurrently. But on the other hand, it is again a bit more dev overhead and totally ok to not do here ;)